beacon-chain/gateway/gateway.go

```diff
@@ -154,8 +154,20 @@ func dial(ctx context.Context, network, addr string) (*grpc.ClientConn, error) {
 
 // dialTCP creates a client connection via TCP.
 // "addr" must be a valid TCP address with a port number.
-func dialTCP(ctx context.Context, addr string) (*grpc.ClientConn, error) {
-	return grpc.DialContext(ctx, addr, grpc.WithInsecure())
+func (g *Gateway) dialTCP(ctx context.Context, addr string) (*grpc.ClientConn, error) {
+	opts := []grpc.DialOption{grpc.WithInsecure()}
+
+	if g.enableDebugRPCEndpoints {
+		// Increase Max call size from 4MB to 16MB to account for end point to retrieve beacon state,
+		// this should be large enough to cover state size up to 300k validators.
+		opts = append(opts, grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(1<<24)))
```

Comment:
Please use a flag for this like we do with the validator.

https://github.com/prysmaticlabs/prysm/blob/546196a6fa8174d68f5c92071ff4bc6edd1ce3d0/validator/flags/flags.go#L37-L40